### ISSUE TITLE
Start-up delay of 2-3min~ due to DB size

### ISSUE THREAD

**Issue by jleeh** [smartcontractkit/chainlink#752]
### System Information

- Network (Main net, Ropsten, etc.): Seen on Main & Ropsten
- Ethereum client (Geth/Parity) and version: Parity 2.1.6 Ropsten & Main
- Go version: Whatever is used in v1.0.12 of the builder
- Host OS: Amazon Linux 2018.03
- Commit (INFO line when starting the node): 3d85b2071cd9824b2f0d70fae052aacfef6330bc (Seen on previous versions)

### Environment Variables
```
LOG_LEVEL: debug
LOG_TO_DISK: true
JSON_CONSOLE: false
ROOT: <redacted>
CHAINLINK_PORT: 6688
CHAINLINK_TLS_PORT: 0
CHAINLINK_TLS_HOST:
ETH_URL: <redacted>
ETH_CHAIN_ID: 1
CLIENT_NODE_URL: <redacted>
TX_MIN_CONFIRMATIONS: 12
TASK_MIN_CONFIRMATIONS: 0
ETH_GAS_BUMP_THRESHOLD: 12
ETH_GAS_BUMP_WEI: 5000000000
ETH_GAS_PRICE_DEFAULT: 20000000000
LINK_CONTRACT_ADDRESS: 0x514910771af9ca656af840dff83e8264ecf986ca
MINIMUM_CONTRACT_PAYMENT: 1.000000000000000000
ORACLE_CONTRACT_ADDRESS:
DATABASE_POLL_INTERVAL: 0s
ALLOW_ORIGINS: https://<redacted>,http://localhost:3000,http://localhost:6688
CHAINLINK_DEV: true
SESSION_TIMEOUT: 15m0s
REAPER_EXPIRATION: 240h0m0s
BRIDGE_RESPONSE_URL:
```

### Additional Information

When I'm upgrading node versions, it triggers the failover process where the standby nodes of the new version are waiting to grab the lock and then start serving.

Although, this process is greatly impacted by the start-up delay that seems to be caused by the size of the bolt db.

Ropsten logs:
```
2018-11-18T12:03:02Z [INFO] Waiting indefinitely for lock on db file <redacted>/db.bolt store/store.go:140
2018-11-18T12:04:08Z [INFO] API exposed for user <redacted> cmd/local_client.go:51
...
2018-11-18T12:07:14Z [DEBUG] Received header #4454339 (0x43f7c3) with hash 0x3599c4330ba5c8cf805beb6449909911e9fa1b2506f45a10b94fc1a90cbada29 services/head_tracker.go:232
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
```
To refer to the timestamps, over 3min before the db is initially grabbed to when the API is exposed through gin.

On checking the size of the DB:
Ropsten = 291M
Main = 305M (Doesn't contain any jobs, just been sat listening to new logs for months)

Same is seen on main:
```
2018-11-18T12:25:58Z [INFO] Waiting indefinitely for lock on db file <redacted>/db.bolt store/store.go:140
2018-11-18T12:26:35Z [INFO] API exposed for user <redacted> cmd/local_client.go:51
...
2018-11-18T12:29:56Z [DEBUG] Environment variables.....
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
```

Thought this could of been related to delay in connecting to the parity instances through the load-balancer, but from the output of that it seems fine:
```
2018-11-18T12:29:55Z [INFO] Connecting to node <redacted> in 1s services/head_tracker.go:205
2018-11-18T12:29:56Z [INFO] Connected to node <redacted> services/head_tracker.go:214
```

When I removed the DB and then re-created the nodes, the node started straight away. As soon as I restored from back-up, it went back to long start-up delays.

My main worry is that in production, when people want to upgrade their nodes and do it as of the docs, they'll be a large delay in the startup of the node, especially if they have failover in multiple sites that includes using NFS disks.

**Comment by dimroc** [comment]
Thanks for the report @jleeh. Can we see the full log for the long boots? I'm curious if migrations are run and how long they take:

```
2018-11-18T12:03:02Z [INFO] Waiting indefinitely for lock on db file <redacted>/db.bolt store/store.go:140
2018-11-18T12:04:08Z [INFO] API exposed for user <redacted> cmd/local_client.go:51
... // Are migrations run here and how long do they take?
2018-11-18T12:07:14Z [DEBUG] Received header #4454339 (0x43f7c3) with hash 0x3599c4330ba5c8cf805beb6449909911e9fa1b2506f45a10b94fc1a90cbada29 services/head_tracker.go:232
[GIN-debug] [WARNING] Running in "d
```

**Comment by jleeh** [comment]
Full log for ropsten:
```
2018-11-18T12:07:21Z [INFO] Waiting indefinitely for lock on db file /db.bolt store/store.go:140
2018-11-18T12:13:55Z [INFO] API exposed for user cmd/local_client.go:51
2018-11-18T12:15:29Z [DEBUG] Fast forwarding to block header #4454373 (0x43f7e5) services/head_tracker.go:287
2018-11-18T12:16:11Z [DEBUG] Tracking logs from last block #4454373 (0x43f7e5) with hash 0x4c14d7f78074bd137bf04a2ba0486bdd897b6e87eea259c8d311c60537bf048e services/head_tracker.go:80
2018-11-18T12:16:11Z [INFO] Connecting to node in 1s services/head_tracker.go:205
2018-11-18T12:16:12Z [INFO] Listening for runlog from block #4454374 (0x43f7e6) for address [all] for job 61fd1abaf93c415ca320f44f965f551e services/subscription.go:217
2018-11-18T12:16:12Z [INFO] Connected to node services/head_tracker.go:214
2018-11-18T12:16:44Z [INFO] ETH Balance for 0x9B19A7A6B6236d189fAD9872C747340F5cF08719: 1.603993110000000000 cmd/local_client.go:102 address=0x9B19A7A6B6236d189fAD9872C747340F5cF08719 balance=1.603993110000000000
2018-11-18T12:16:44Z [INFO] Link Balance for 0x9B19A7A6B6236d189fAD9872C747340F5cF08719: 0.000000000000000000 cmd/local_client.go:105 address=0x9B19A7A6B6236d189fAD9872C747340F5cF08719 balance=0.000000000000000000
2018-11-18T12:16:44Z [DEBUG] Environment variables...
[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
```
I've only really seen it run migrations when its been started new since 0.5.0.

**Comment by jleeh** [comment]
Progress on this:

- Increased the specs of both RAM/CPU and disk on the containers and no perf improvement was seen.
- Changed the network disks to High IO mode and no perf improvement was seen.
- When running locally in Docker, it was slightly more performant but similar delays were seen.
- When running outside of Docker, performance gains were seen of around 60s on start.

Next step is to create nodes in AWS that run outside of containers to see if similar gains are seen. Although, I'm focusing my time on getting ELK metrics properly configured so I may be a bit delayed in doing that.

**Comment by se3000** [comment]
@jleeh I believe this has been resolved with the version 0.6. Please reopen if not.
